client/src/NeedFormGoods.tsx

```diff
@@ -0,0 +1,227 @@
+import type { Theme } from '@material-ui/core/styles';
+import { Button } from '@material-ui/core';
+import { CustomTextField, CustomRadio, CustomSelect, CustomFileUpload } from './FormElements';
+
+const categories = [
+    { value: 'figs', text: 'Figs' },
+    { value: 'peaches', text: 'Peaches' },
+    { value: 'pears', text: 'Pears' },
+];
+const conditions = [
+    { value: 'like-new', text: 'Like New' },
+    { value: 'excellent', text: 'Excellent' },
+    { value: 'good', text: 'Good' },
+];
+const needTypes = [
+    { value: 'donation', text: 'Donation' },
+    { value: 'short-term', text: 'Short term loan (<1 month)' },
+    { value: 'long-term', text: 'Long term loan (>1 month)' },
+];
+const deliveryTypes = [
+    { value: 'pick-up', text: 'Pick up only' },
+    { value: 'drop-off', text: 'Drop off only' },
+    { value: 'pick-up-drop-off', text: 'Pick up and drop off' },
+];
+
+const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+        '& .MuiFormLabel-root': {
+            color: theme.palette.text.primary,
+            marginTop: '1rem',
+            marginBottom: '14px',
+        },
+        '& .MuiOutlinedInput-root': {
+            '& fieldset': {
+                border: `1px solid ${theme.custom.form.borderColor}`,
+                borderRadius: theme.custom.form.borderRadius,
+            },
+        },
+        '& .MuiInputBase-input': {
+            color: theme.palette.text.primary,
+        },
+        '& .MuiFormControl-root': {
+            width: '100%',
+        },
+        '& .MuiRadio-colorSecondary.Mui-checked': {
+            color: theme.palette.text.primary,
+        },
+        '& .MuiSelect-root em': {
+            color: theme.palette.text.secondary,
+        },
+        '& .MuiRadio-root': {
+            padding: '2px 9px',
+        },
+    },
+    formBox: {
+        width: '100%',
+        padding: '0 2rem 4rem 2rem',
+        marginTop: '3rem',
+        '& h2': {
+            width: '100%',
+        },
+    },
+    borderBox: {
+        width: '100%',
+        border: '1px solid #C4C4C4',
+        padding: '4rem 7rem',
+        boxSizing: 'border-box',
+    },
+    submitbuttons: {
+        justifyContent: 'center',
+        '& .MuiButton-root': {
+            margin: '0.5rem 1rem',
+            borderRadius: '10px',
+        },
+    },
+}));
+
+interface ShareANeedData {
+    title: string;
+    location: string;
+    description: string;
+    category: string;
+    condition: string;
+    quantity: string;
+    needType: string;
+    deliveryMethod: string;
+}
+
+const initialFormData: ShareANeedData = {
+    title: '',
+    location: '',
+    description: '',
+    category: '',
+    condition: '',
+    quantity: '',
+    needType: '',
+    deliveryMethod: '',
+};
+
+function NeedForm() {
+    const classes = useStyles();
+
+    const [ formData, setFormData ] = React.useState(initialFormData);
+
+    const handleChange = (evt: React.ChangeEvent<HTMLInputElement>): void => {
```

Comment:
got TS here on compile
for the typing of this function and where it's used down the pipeline
let's just follow MUI docs https://v4.mui.com/components/selects/ for now (despite the `unknown`)
(see one of the snippets and then select TS)